packages/types/src/schemas/common.ts

```diff
@@ -161,8 +161,32 @@ export const terminalSchema = z.union([
   }),
 ]);
 
+export const editorSchema = z.union([
+  // can either be completely removed by setting it to `false`
+  z.boolean().optional(),
+
+  z.strictObject({
+    fileTree: z
+      .union([
+        // or you can only remove the file tree
+        z.boolean(),
+
+        // or configure file tree with options
+        z.strictObject({
+          allowEdits: z
```

Comment:
I think we want something more fine grained where you specify in which scope you want to allow files and folder to be added.

Reply:
Sounds good! Though this is something that could be implemented after this first PR without breaking the API too, so that `true` would mean "allow everything". 

But I can also take a look at this once I've addressed other review findings. 👍